### PR TITLE
SPMI: Return recorded base instrumentation data address

### PR DESCRIPTION
**Pull request by jakobbotsch**
This comment has not been accurate since the switch to the schema-based PGO format. Today this buffer records only instrumentation data and only the generated code accesses it, so we can return the recorded address and fix one cause of spurious SPMI diffs.

### CODE REVIEW

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib PTAL @AndyAyersMS

@BruceForstall Turns out this does not need any SPMI update since we were already recording this address.

**Comment by jakobbotsch** [comment]
I will fix #74718 separately, for that one we will need to do a small part of the schema layout inside SPMI when there is schema mismatch.

File: src/coreclr/tools/superpmi/superpmi-shared/methodcontext.cpp [hunk 1]
```diff
@@ -5645,19 +5645,9 @@ HRESULT MethodContext::repAllocPgoInstrumentationBySchema(
...
-    // Todo, perhaps: record the buffer as a compile result instead, and defer copying until
-    // jit completion so we can snapshot the offsets the jit writes.
-    //
-    // Add 16 bytes of represent writeable space
-    size_t bufSize = maxOffset + 16;
-    *pInstrumentationData = (BYTE*)AllocJitTempBuffer(bufSize);
-    cr->recAddressMap((void*)value.instrumentationDataAddress, (void*)*pInstrumentationData, (unsigned)bufSize);
+    // We assume JIT does not write or read from this buffer, only generate
+    // code that uses it.
+    *pInstrumentationData = (BYTE*)value.instrumentationDataAddress;
```
Comment (AndyAyersMS) [inline src/coreclr/tools/superpmi/superpmi-shared/methodcontext.cpp:5650]: You can also delete the `maxOffset` local and the computation above that determines its value.

**Review by AndyAyersMS — APPROVED** [review]

**Comment by BruceForstall** [comment]
Thanks for doing this.

**Pull request merged**